src/unix/netbsd.c

```diff
@@ -79,19 +74,19 @@ int uv_exepath(char* buffer, size_t* size) {
   mib[1] = KERN_PROC_ARGS;
   mib[2] = -1;
   mib[3] = KERN_PROC_PATHNAME;
-  int_size = ARRAY_SIZE(int_buf);
+  saved_errno = errno;
+  errno = 0;
 
-  if (sysctl(mib, 4, int_buf, &int_size, NULL, 0))
-    return UV__ERR(errno);
-
-  /* Copy string from the intermediate buffer to outer one with appropriate
-   * length.
-   */
-  /* TODO(bnoordhuis) Check uv__strscpy() return value. */
-  uv__strscpy(buffer, int_buf, *size);
+  if (sysctl(mib, ARRAY_SIZE(mib), buffer, size, NULL, 0)) {
+    if (errno != ENOSPC)
+      return UV__ERR(errno);
```

Comment:
This doesn't restore `errno`. :-)

That said, as a rule we don't care about clobbering errno.

Reply:
In the error path we might not care.

Reply:
Well originally I wanted simplification I got the opposite :-)